project/shih-chuan/README.md

```diff
@@ -0,0 +1,95 @@
+
+    ```pseudo
+    var endpoints;   # list of endpoints, sorted by angle
+    var open = [];   # list of walls the sweep line intersects
+
+    loop over endpoints:
+        remember which wall is nearest
+        add any walls that BEGIN at this endpoint to 'walls'
+        remove any walls that END at this endpoint from 'walls'
+
+        figure out which wall is now nearest
+        if the nearest wall changed:
+            fill the current triangle and begin a new one
+    ```
+
+## Prospective users
+
+* Game Developer: Determine whether something in a game world is visible for a character is very common in game developing. There are even games making this as one of their main features, like [Monaco](https://store.steampowered.com/app/113020/Monaco_Whats_Yours_Is_Mine/?l=tchinese) and [Among us](https://store.steampowered.com/app/945360/Among_Us/)
+
+| ![among-us](./assets/among-us.jpeg) | ![monaco](./assets/monaco.jpg) |
+|:-----------------------------------:|:-----------------------------------:|
+| **Among us** |**Monaco**|
+
+* House planning for Wireless Network: Wi-Fi signals are most weakened by having to go through thick walls. To avoid signals from being weakened by the walls, this system may provide some insights to plan the wireless network in your house.
+
+## System Architecture
+
+The system should provide a graphical user interface allow users to design a 2D map, place walls and move a light source to play around. Additionally, when the user is interacting with any objects on the map, the system needs to highlight the area reachable from the light source on the map in real time.
+
+| ![among-us](./assets/system_arch.jpg) |
+|:-----------------------------------:|
+| **System flow chart** |
+
+## API Description
+
+* Python API:
+  * `addWall(start, end)`: create a wall on the map and return its id
+  * `rmWall(wall_id)`: remove the wall by its id
+  * `moveLightSource(dest)`: move light source to the destination coordinate
+  * `lightArea(walls, light_source)`: call the c++ api and show the lit area on the map
+
+* C++ API:
+  * `lightArea(walls, light_source)`: calculate all the areas where lights from a given single light source can reach on a 2D map and return the coordinates of the lit places and its area.
+
+## Engineering Infrastructure
+
+* Automatic build system: `cmake`
+* Version control: `git`
+* Testing framework: `pytest`
+* Documentation: GitHub `README.md`
+
+## Schedule
+
+Planning phase (6 weeks from 9/19 to 10/31)
+
+Week 1 (10/31): setup the environment and get familiar with the algorithm
+
+Week 2 (11/7): Implement the algorithm with C++
+
+Week 3 (11/14): Finish C++ and bind to python
```

Comment:
It may be more productive to make the wrapper alone with C++.  Python code is easier to test and it will save you time.

Reply:
LGTM